.github/workflows/ci.yml

```diff
@@ -0,0 +1,51 @@
+name: CI
+
+on:
+  push:
+  pull_request:
+
+# TODO: do we need this?
+# permissions:
+#   contents: read # for actions/checkout to fetch code
```

Comment:
afaik this is pretty much implied. most workflows checkout without specifying this anyway.